projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -435,7 +435,7 @@ public static function get_standalone_info() {
 	public static function has_paid_plan_for_product() {
 		// First check site features (if there's a feature that identifies the paid plan)
 		if ( static::$feature_identifying_paid_plan ) {
-			if ( static::does_site_have_feature( static::$feature_identifying_paid_plan ) ) {
+			if ( Wpcom_Products::current_plan_supports( static::$feature_identifying_paid_plan ) ) {
```

Comment:
My initial thought is that this might break some things unintentionally since the feature check is meant to take into account all purchases on a site and `current_plan_supports` is only going to check against the current plan (not add on purchases).

I am wondering if there's a different place we can fix the issue 🤔

Reply:
@jboland88, Yeah, I believe `static::does_site_have_feature( static::$feature_identifying_paid_plan )` and `Wpcom_Products::current_plan_supports( static::$feature_identifying_paid_plan )` are both comparing the `$feature_identifying_paid_plan` to the same set of site features (the sites 'active' features).

The previous way (`static::does_site_have_feature()`) pulls the sites 'active' features from /rest/v1.1/sites/229710838/features

The Current_Plan way pulls the sites 'active' features from /rest/v1.1/sites/229710838 and from the response data: plan->features->active

But they're basically both doing the same thing (comparing the same things), to my understanding.

Reply:
Ok, after a Slack discussion: p1738263782787889-slack-C07RQ9B3UFK, we're going to take a slightly different approach:  Like this:

```php
/**
 * Gets the site's currently active "plan" (bundle).
 *
 * @return array
 */
public static function get_site_current_plan( $reload = false ) {
	static $reloaded_already = false;

	if ( $reload && ! $reloaded_already ) {
		Current_Plan::refresh_from_wpcom();
		$reloaded_already = true;
	}

	return Current_Plan::get();
}
```